loadNeustarReputationIntoES.js

```diff
@@ -0,0 +1,1288 @@
+var http = require('http');
+var program = require('commander');
+var makeSource = require("stream-json");
+var winston = require('winston');
+const fs = require('fs');
+const fileName = '/tmp/neustarBulkUploadResponse.log';
+program
+    .version('0.0.1')
+    .option('-s, --skiplines <n>', 'skip first n lines', parseInt)
+    .option('-b, --batchsize <n>', 'upload in bulk size', parseInt)
+    .option('--host [esHost]', 'elasticSearch Host')
+    .option('--port [esPort]', 'elasticSearch port')
+    .option('--deleteAll', 'delete all indexes except monolith one')
+    .option('--deleteOld', 'delete indexes older that whats used by monolith')
+    .option('--deleteSingle [indexToDelete]', 'delete single index')
+    .option('--upload', 'upload csv file')
+    .option('--switch', 'start using new index for ip reputation')
+    .option('--switchToLastUsedIndex', 'start using new index for ip reputation')
+    .option('--logLevel [winstonLogLevel]', 'log level for console logging')
+    .option('--logBulkUpload <n>', 'write BulkUploadResponse to log file and cap its size', parseInt)
+    .parse(process.argv);
+
+//Regular expression used when getting all Neustar IP reputation indexes on ES
+const indexRegExp = "neustar.ipinfo.*";
+const metadataIndexName = "neustar.metadata";
+const scratchSpaceIndexName = "neustar.scratch.space";
+const metadataTypeName = "1";
+const objectType = "1";//name of mapping for Neustar index
+
+//all ip info indexes will have this prefix
+//the suffix is the timestamp (int measured from epoch)
+const neustarIndexPrefix = "neustar.ipinfo.";
+var recordCount = 0;
+var total = 0;
+
+setLogLevel();
+bulkUploadLogCheck();
+
+function setLogLevel() {
+    if (program.logLevel == null) {
+        winston.level = 'info';
+    } else {
+        switch (program.logLevel.toString()) {
+            case 'debug':
+            case 'verbose':
+                winston.level = program.logLevel.toString();
+                break;
+            default:
+                winston.level = 'info';
+                break;
+        }
+    }
+    console.log("Logging level set to " + winston.level);
+}
+
+function bulkUploadLogCheck() {
+    //check for logFile exist
+    if (program.logBulkUpload != null) {
+        fs.writeFile(fileName, '', {flag: 'w'}, function (err) {
+            if (err) {
+                throw err;
+            }
+            winston.log('info', "Created log file");
+            optionCheck();
+        });
+    } else {
+        optionCheck();
+    }
+}
+
+function countNonNull(ary) {
+    return ary.filter(function(elem) { return elem != null; }).length;
+}
+
+function optionCheck() {
+    var startDateTime = new Date();
+    winston.log('info', "Time started: " + " hours " + startDateTime.getUTCHours() +
+        " minutes " + startDateTime.getUTCMinutes() + " seconds "  + startDateTime.getUTCSeconds());
+    if (program.host == null || program.port == null || program.host == true || program.port == true) {
```

Comment:
So program.host and port are not boolean but strings. What i have noticed with the commander module, is that if we use that option but don't give it a parameter. It will get teh value true.
That is:
node helloWorld.js //this means host parameter is null
node helloWorld.js --host  //this means value of host parameter is true
node helloWorld.js --host localhost  //this means value of host parameter is localhost